From ae10c15867d9a5b85eefaf11333bd30473af4b2b Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Sat, 2 May 2020 13:43:06 +0100
Subject: [PATCH] overlays: gpio-keys: Avoid open-drain warnings

The i2c-gpio driver expects to use a GPIO in open-drain mode. Failure
to configure it in that way causes alarming warnings in the kernel log.
The BCM283x and BCM2711 GPIO blocks don't support open-drain mode,
but i2c-gpio works anyway. Silence the warning by declaring that
open-drain mode has been enabled by other means.

See: https://github.com/raspberrypi/firmware/issues/1381

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts | 2 ++
 1 file changed, 2 insertions(+)

--- a/arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts
@@ -16,6 +16,8 @@
 					 &gpio 24 0 /* scl */
 					>;
 				i2c-gpio,delay-us = <2>;        /* ~100 kHz */
+				i2c-gpio,sda-open-drain;
+				i2c-gpio,scl-open-drain;
 				#address-cells = <1>;
 				#size-cells = <0>;
 			};
